app/main.py

```diff
@@ -1 +1,28 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, Hidden: {self.hidden}}}")
+
+    def check_alive(self) -> None:
+        if self.health <= 0 and self in Animal.alive:
```

Comment:
The `check_alive` method correctly checks if the animal's health is less than or equal to zero and removes it from the `alive` list if it is. Ensure that this logic is consistently applied whenever an animal's health changes.